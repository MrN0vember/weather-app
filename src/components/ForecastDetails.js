import React from "react";
import PropTypes from "prop-types";
import moment from "moment";

const ForecastDetails = ({ forecast }) => {
  const { date, temperature, wind, humidity } = forecast;
  return (
    <div className="forecast-details">
      <p className="forecast-details_date">
        {moment(date).format("ddd Do MMM")}
      </p>
      <p className="forecast-details_tempreature">
        <b> Max Tempreature:</b>
        {temperature.max}
        &deg;C
      </p>
      <p className="forecast-details_tempreature">
        <b> Min Tempreature:</b>
        {temperature.min}
        &deg;C
      </p>
      <p className="forecast-details_humidity">
        <b>Humidity:</b>
        {`${humidity}%`}
      </p>
      <p className="forecast-details_wind">
        <b>Wind:</b>
        {`${wind.speed}mph`}
      </p>
    </div>
  );
};

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    humidity: PropTypes.number,
    temperature: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }),
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }),
  }).isRequired,
};

export default ForecastDetails;
