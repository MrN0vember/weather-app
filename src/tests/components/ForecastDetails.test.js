import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  it("renders correctly", () => {
    const validProps = {
      date: 1111111,
      temperature: {
        min: 8,
        max: 17,
      },
      wind: {
        speed: 3,
        direction: "e",
      },
      humidity: 50,
    };
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
