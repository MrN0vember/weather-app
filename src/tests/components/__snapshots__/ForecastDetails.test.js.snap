// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <p
      class="forecast-details_date"
    >
      Thu 1st Jan
    </p>
    <p
      class="forecast-details_tempreature"
    >
      <b>
         Max Tempreature:
      </b>
      17°C
    </p>
    <p
      class="forecast-details_tempreature"
    >
      <b>
         Min Tempreature:
      </b>
      8°C
    </p>
    <p
      class="forecast-details_humidity"
    >
      <b>
        Humidity:
      </b>
      50%
    </p>
    <p
      class="forecast-details_wind"
    >
      <b>
        Wind:
      </b>
      3mph
    </p>
  </div>
</DocumentFragment>
`;
